patches/net/minecraft/world/item/Item.java.patch

```diff
@@ -12,11 +12,20 @@
      public static final ResourceLocation BASE_ATTACK_DAMAGE_ID = ResourceLocation.withDefaultNamespace("base_attack_damage");
      public static final ResourceLocation BASE_ATTACK_SPEED_ID = ResourceLocation.withDefaultNamespace("base_attack_speed");
      public static final int DEFAULT_MAX_STACK_SIZE = 64;
-@@ -89,12 +_,13 @@
+@@ -71,6 +_,7 @@
+     @Nullable
+     private String descriptionId;
+     private final FeatureFlagSet requiredFeatures;
++    private final java.util.Set<net.neoforged.neoforge.flag.Flag> requiredFlags;
+ 
+     public static int getId(Item p_41394_) {
+         return p_41394_ == null ? 0 : BuiltInRegistries.ITEM.getId(p_41394_);
+@@ -89,12 +_,14 @@
          this.components = p_41383_.buildAndValidateComponents();
          this.craftingRemainingItem = p_41383_.craftingRemainingItem;
          this.requiredFeatures = p_41383_.requiredFeatures;
 -        if (SharedConstants.IS_RUNNING_IN_IDE) {
++        this.requiredFlags = java.util.Collections.unmodifiableSet(p_41383_.requiredFlags);
```

Comment:
I'd recommend using `Set.copyOf` here; it will have a smaller memory footprint for single/dual sets, and use a canonical empty instance. This will avoid the memory impact of storing a separate set for every item in most cases.